app/grandchallenge/participants/forms.py

```diff
@@ -0,0 +1,30 @@
+from django.forms import ModelForm, TextInput
+
+from grandchallenge.core.forms import SaveFormInitMixin
+from grandchallenge.core.widgets import JSONEditorWidget
+from grandchallenge.participants.models import RegistrationQuestion
+
+
+class RegistrationQuestionForm(SaveFormInitMixin, ModelForm):
+
+    class Meta:
+        model = RegistrationQuestion
+        fields = (
+            "question_text",
+            "question_help_text",
+            "required",
+            "schema",
+        )
+        widgets = {
+            "question_text": TextInput,
+            "question_help_text": TextInput,
+            "schema": JSONEditorWidget(),
+        }
+
+    def __init__(self, *args, challenge, **kwargs):
+        super().__init__(*args, **kwargs)
+        if (
+            not hasattr(self.instance, "challenge")
+            or not self.instance.challenge
+        ):
+            self.instance.challenge = challenge
```

Comment:
Seems a bit of a strange construction. What if the challenges differ?

Reply:
It is strange. Reusing the form for both creating and updating. I've it to be more clearly a check to only update the challenge if it is being created.

If they differ: the model validation should pick that up.